### PR TITLE
feat: TS example includes getServerSideProps

### PR DESCRIPTION
**Pull request by Vandivier**
1. extends the TS example to address the case of getServerSideProps. Credit to @benhickson for the approach (ref: https://github.com/vvo/next-iron-session/issues/298#issuecomment-856759388)
2. add User and UserProps types to generally strengthen typing in the TS example

### CODE REVIEW

**Comment by emipc** [comment]
I had to export this type in order to use it in `getServerSideProps` or I was getting TS errors complaining about implicit `any` types for `req` and `res`:

```ts
export type NextIronGetServerSidePropsContext = {
  req: NextIronRequest;
  res: NextApiResponse;
};
```

File: examples/next-typescript/pages/profile-ssr.tsx [hunk 1]
```diff
@@ -3,7 +3,19 @@ import Layout from "../components/Layout";
 import withSession from "../lib/session";
 import PropTypes from "prop-types";
 
-const SsrProfile = ({ user }) => {
+type User = {
```
Comment (vvo) [inline examples/next-typescript/pages/profile-ssr.tsx:6]: In a real world application, wouldn't this type go into a folder like `@customTypes/User.ts`? And then `import type { User } from "@customTypes/User"`?

Or something similar. The idea being to have the type declared in a single file and import it everywhere we need it (profile ssr, userUser).

Comment (Vandivier) [reply examples/next-typescript/pages/profile-ssr.tsx]: done!

**Comment by vvo** [comment]
Hey @Vandivier I completely rewrote the library in TypeScript, now the types are good I believe, let me know: https://github.com/vvo/iron-session/releases/tag/v6.0.0 + https://github.com/vvo/iron-session#typing-session-data-with-typescript

**Pull request closed without merging**